zjit/src/hir.rs

```diff
@@ -2579,6 +2606,7 @@ impl Function {
     pub fn rpo(&self) -> Vec<BlockId> {
         let mut result = self.po_from(self.entry_block);
         result.reverse();
+        result.insert(1, self.jit_entry_block); // Put jit_entry_block after entry_block
```

Comment:
Why do we need to do this if the interpreter entry jumps to the jit entry?

Reply:
From other comments in this PR, I'm not sure if you understand the intended relationship between blocks.

It's implemented as:

```
bb0 (entry_block) ---→ bb2 (first_block, main CFG)
                       ↑
bb1 (jit_entry_block) -┘
```

Your understanding seems to be:

```
bb0 (entry_block)      bb2 (first_block, main CFG)
      ↓                ↑
bb1 (jit_entry_block) -┘
```

which is not the case.

Because there are two ways to enter JIT code, retrieving the graph from one entry is not enough.